courses/utils.py

```diff
@@ -0,0 +1,301 @@
+"""
```

Comment:
Take or leave: every function in here apart from `is_program_text_id` is not a utility function IMO and should go into `courses/api.py` or some more specific `<whatever>_api.py` file

Reply:
I moved everything here that wasn't certificate related (and hence getting deleted) into `api.py`.